src/klein/_decorators.py

```diff
@@ -57,17 +67,20 @@ def decorator(wrapper):
 
 
 def named(name):
+    # type: (Text) -> Callable[[C], C]
```

Comment:
Since `name` is converted to `str` later, it would be safer to use that as the argument type as well.

Reply:
I think the intention here in Python 2 is that either is an acceptable input, and changing this would be sort of an API change.

When we jettison Python 2, the conversion, and this distinction, becomes unnecessary.

Reply:
That is, I don't think that `named("foo")` should raise a typing error in Py2.